### PR TITLE
Simplify `F.get_item` test

### PR DESCRIPTION
**Pull request by niboshi**
Part of #6071 

There were no double-backward test for advanced indexing.
This PR enables such tests.

ChainerX support needs further fix (#6986, PR: #6991).

### CODE REVIEW

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
py2 tests seem to fail in the case of `slice.shape == (0,)`.

**Comment by asi1024** [comment]
Jenkins, test this please.

File: tests/chainer_tests/functions_tests/array_tests/test_get_item.py [hunk 1]
```diff
@@ -10,179 +10,148 @@
...
+
+    def _convert_slices(self, slices, device):
+        # Converts advanced indexing slices (of numpy.ndarray) to respective
+        # backend arrays.
+        if isinstance(slices, list):
+            return [self._convert_slices(a, device) for a in slices]
+        if isinstance(slices, tuple):
+            return tuple([self._convert_slices(a, device) for a in slices])
+        if isinstance(slices, numpy.ndarray):
+            return device.send(slices)
```
Comment (hvy) [inline tests/chainer_tests/functions_tests/array_tests/test_get_item.py]: This PR changes the test to always use slices with the same "xp" when `slices` is an ndarray. It used to always use slices of `np.ndarray`. Both seems to be supported.

File: tests/chainer_tests/functions_tests/array_tests/test_get_item.py [hunk 2]
```diff
@@ -10,179 +10,148 @@
...
-     {'slices': (slice(None), numpy.array([True, False, True])),
-         'sliced_shape': (4, 2, 2)},
-     {'slices': numpy.array([False, False, False, False]),
-         'sliced_shape': (0, 3, 2)},
-     {'slices': (3, 2, Ellipsis, 1),
-         'sliced_shape': ()},
-     {'slices': (numpy.array(False)),
-         'input_shape': (), 'sliced_shape': (0,)},
-     {'slices': (numpy.array(True)),
-         'input_shape': (), 'sliced_shape': (1,)},
```
Comment (hvy) [inline tests/chainer_tests/functions_tests/array_tests/test_get_item.py]: Could you add the `inputs_shape` back?

Comment (asi1024) [reply tests/chainer_tests/functions_tests/array_tests/test_get_item.py]: Fixed.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Failures seem unrelated. LGTM!

**Pull request merged**